src/libraries/System.Formats.Tar/src/System/Formats/Tar/TarHeader.Read.cs

```diff
@@ -425,6 +432,30 @@ TarEntryType.RenamedOrSymlinked or
             return header;
         }
 
+        // Calculates the checksum for a TAR header by summing all bytes,
+        // but treating the checksum field as if it contained spaces (ASCII 32)
+        private static int CalculateHeaderChecksum(ReadOnlySpan<byte> buffer)
+        {
+            int calculatedChecksum = 0;
+
+            // Process bytes before the checksum field
+            for (int i = 0; i < FieldLocations.Checksum; i++)
+            {
+                calculatedChecksum += buffer[i];
+            }
+
+            // For the checksum field, treat all bytes as spaces (ASCII 32)
+            calculatedChecksum += (byte)' ' * FieldLengths.Checksum;
+
+            // Process bytes after the checksum field
+            for (int i = FieldLocations.Checksum + FieldLengths.Checksum; i < TarHelpers.RecordSize; i++)
+            {
+                calculatedChecksum += buffer[i];
+            }
```

Comment:
Consider the following, based on perf test results.  I know at one point the JIT would produce better code when it could elide bounds checks for an array/span.

```suggestion
            for (int i = 0; i < buffer.Length; i++)
            {
                if (i == FieldLocations.Checksum)
                {
                    calculatedChecksum += (byte)' ' * FieldLengths.Checksum;
                    i += FieldLengths.Checksum;
                    continue;
                } 
                calculatedChecksum += buffer[i];
            }
```